pvanalytics/quality/data_shifts.py

```diff
@@ -0,0 +1,306 @@
+        and power data streams. This series represents the summed daily values
+        of the particular data stream.
+
+    Returns
+    -------
+    Pandas series with a datetime index:
+        Time series, after data processing. This includes min-max
+        normalization, and, if the time series is in greater than 2 years in
+        length, seasonality removal.
+    """
+    # Convert the time series to a dataframe to do the pre-processing
+    column_name = time_series.name
+    if column_name is None:
+        column_name = "value"
+        time_series = time_series.rename(column_name)
+    df = time_series.to_frame()
+    # Min-max normalize the series
+    df[column_name + "_normalized"] = (df[column_name] -
+                                       df[column_name].min())\
+        / (df[column_name].max() - df[column_name].min())
+    # Check if the time series is greater than one year in length. If not, flag
+    # a warning and pass back the normalized time series
+    if not remove_seasonality:
+        return df[column_name + "_normalized"]
+    else:
+        # Take the median of every day of the year across all years in the
+        # data, and use this as the seasonality of the time series
+        df['month'] = pd.DatetimeIndex(df.index).month
+        df['day'] = pd.DatetimeIndex(pd.Series(df.index)).day
+        df['seasonal_val'] = df.groupby(['month',
+                                         'day'])[column_name +
+                                                 "_normalized"].transform(
+                                                     "median")
+        # Remove seasonlity from the time series
+        return df[column_name + "_normalized"] - df['seasonal_val']
+
+
+def detect_data_shifts(time_series, filtering=True, use_default_models=True,
+                       method=rpt.BottomUp, cost='rbf', penalty=40):
+    """
+    Detect data shifts in the time series, and return list of dates where these
+    data shifts occur.
+
+    Parameters
+    ----------
+    time_series : Pandas series with datetime index.
+        Daily time series of a PV data stream, which can include irradiance
+        and power data streams. This series represents the summed daily values
+        of the particular data stream.
+    filtering : Boolean.
+        Whether or not to filter out outliers and stale data from the time
+        series. If True, then this data is filtered out before running the
+        data shift detection sequence. If False, this data is not filtered
+        out. Default set to True.
+    use_default_models: Boolean.
+        If set to True, then default CPD model parameters are used, based
+        on the length of the time series (Window-based models for time series
+        shorter than 2 years in length and BottomUp models for time series
+        longer than 2 years in length). If set to True, none of the method +
+        cost + penalty variables are used.
+    method: ruptures search method object.
```

Comment:
I would prefer default to None. Same for `cost`.
```suggestion
    method: ruptures search method instance or None, default None.
```